src/extension.ts

```diff
@@ -57,11 +57,14 @@ export async function activateInternal(context: vscode.ExtensionContext, perfSta
         context.subscriptions.push(ext.appResourceTree.trackTreeItemCollapsibleState(ext.appResourceTreeView));
 
         // Hook up the resolve handler
-        registerEvent('treeItem.expanded', ext.appResourceTree.onDidExpandOrRefreshExpandedTreeItem, async (context: IActionContext, treeItem: AzExtTreeItem) => {
+        registerEvent('treeItem.expanded', ext.appResourceTree.onDidExpandOrRefreshExpandedTreeItem, async (context: IActionContext, data: OnDidExpandOrRefreshExpandedEmitterData) => {
             context.errorHandling.suppressDisplay = true;
-
-            if (treeItem instanceof GroupTreeItemBase) {
-                await treeItem.resolveAllChildrenOnExpanded(context);
+            if (data.item instanceof GroupTreeItemBase) {
+                if (data.source !== 'refresh') {
```

Comment:
Don't call `refreshUIOnly` if this event was triggered from a refresh to prevent an infinite loop.